superset/db_engine_specs.py

```diff
@@ -737,12 +761,20 @@ def handle_cursor(cls, cursor, query, session):
         while polled:
             # Update the object and wait for the kill signal.
             stats = polled.get('stats', {})
-
             query = session.query(type(query)).filter_by(id=query.id).one()
             if query.status in [QueryStatus.STOPPED, QueryStatus.TIMED_OUT]:
                 cursor.cancel()
                 break
 
+            if (
+                config.get('PREFETCH_ASYNC') and
+                (not query.has_loaded_early)
+            ):
+                query.has_loaded_early = True
+                limit = config.get('PREFETCH_ROWS')
+                PrestoEngineSpec.prefetch_results(
```

Comment:
Is this possible that since `fetch_many` is synchronous, it will prevent publishing any query % progress until some rows are fetched? I think we may be loosing or obfuscating progress here. Say a query with a large groupby returning a small result set that takes 5 minutes to scan will show 0% until it can return the small result set. Now I also wonder how the UI does if/when the prefetch and final result occur right at around the same moment (large scan query with small result set), does it flicker, does it look ok?